docs/tutorials/provider/README.md

```diff
@@ -1,86 +1,179 @@
 
 ### 1.1 Define Digital Twin Provider Interface
 
 A digital twin provider needs an interface. The interface will expose operations that allow digital twin consumers to access a subset of in-vehicle signals that your digital provider makes available.
 
->Tip: A suggested approach to defining your digital twin provider is to adopt the perspective of a digital twin consumer. This involves consideration of the operations and their corresponding names. For example, the [digital twin provider sample interface](../../../samples/interfaces/sample_grpc/v1/digital_twin_provider.proto), the specified operations are `Subscribe`, `Unsubscribe`, `Get`, `Set`, `Invoke` and `Stream`.
+>Tip: A suggested approach to defining your digital twin provider is adopt the perspective of a digital twin consumer. This requires consideration of the operations and their corresponding names to interact with each in-vehicle signal and command. For example, for the [digital twin provider sample interface](../../../samples/interfaces/sample_grpc/v1/digital_twin_provider.proto), the specified operations are `Subscribe`, `Unsubscribe`, `Get`, `Set`, `Invoke` and `Stream`.
 
-In this section, you will utilize the [digital twin provider sample interface](../../../samples/interfaces/sample_grpc/v1/digital_twin_provider.proto). Specifically, you will use the `Subscribe`, `Set` and `Invoke` operations from this interface.
-
->Please note that this interface serves as an example of what a digital twin provider's interface could look like. Feel free to replicate these operation names, modify them, or even add new ones as per your requirements.
+The [digital twin provider sample interface](../../../samples/interfaces/sample_grpc/v1/digital_twin_provider.proto) serves as an example of what a digital twin provider's interface could look like. Feel free to replicate these operation names, modify them, or even add new ones as per your requirements. These operations are non-prescriptive. It is up to the developers of the in-vehicle digital twin to come up with their own convention.
 
 #### Sample Digital Twin Provider Interface
 
-1. Consider the in-vehicle signals `ambient air temperature` and `is air conditioning active`, as well as the command `show notification` that you defined in the [Tutorial: Create an In-Vehicle Model with DTDL](../in_vehicle_model/README.md).
+This section provides an example of a digital twin provider interface. To reiterate, you are free to use this interface as a starting point or you may come up with your own convention.
+
+1. Consider the in-vehicle signals *ambient air temperature* and *is air conditioning active*, as well as the command *show notification* that you defined in the [Tutorial: Create an In-Vehicle Model with DTDL](../in_vehicle_model/README.md).
+
+1. Reference the digital twin provider sample interface below which is defined using a protobuf file:
+
+  ```proto
```

Comment:
Including code snippets like this and the rust ones later on means less clicks for people reading the doc, but it also means that it's very easy for this to diverge if the code it's referencing ever changes. I'm not sure which is more preferable and it seems opinions might be divided in this PR, but we should consider whether the convenience of having code copied into readmes is worth the maintenance cost. Having tutorials stop working because someone forgot to update the code snippet is not a great scenario

Reply:
I've added new sample code for this tutorial. The tutorial code is tied to this document, so if the tutorial code changes then this document may need to be potentially updated.


I've removed copying the contents of a protobuf file like I've done in this past commit, and instead I referred the user to the specific proto file